flepimop/gempyor_pkg/src/gempyor/parameters.py

```diff
@@ -118,22 +152,65 @@ def __init__(
         logging.debug(f"NPI overlap operation is {self.stacked_modifier_method} ")
 
     def picklable_lamda_alpha(self):
-        """These two functions were lambda in __init__ before, it was more elegant. but as the object needs to be pickable,
-        we cannot use second order function, hence these ugly definitions"""
+        """
+        Read the `alpha_val` attribute.
+        
+        This defunct method returns the `alpha_val` attribute of this class which is
+        never set by this class. If this method is called and the `alpha_val` attribute
+        is not set an AttributeError will be raised.
+        
+        Returns:
+            The `alpha_val` attribute.
+        """
         return self.alpha_val
 
     def picklable_lamda_sigma(self):
+        """
+        Read the `sigma_val` attribute.
+        
+        This defunct method returns the `sigma_val` attribute of this class which is
+        never set by this class. If this method is called and the `sigma_val` attribute
+        is not set an AttributeError will be raised.
+        
+        Returns:
+            The `sigma_val` attribute.
+        """
         return self.sigma_val
 
     def get_pnames2pindex(self) -> dict:
+        """
+        Read the `pnames2pindex` attribute.
+        
+        This redundant method returns the `pnames2pindex` attribute of this class.
+        
+        Returns:
+            A mapping parameter names to their location in the `pnames` attribute.
+        """
         return self.pnames2pindex
 
     def parameters_quick_draw(self, n_days: int, nsubpops: int) -> ndarray:
         """
-        Returns all parameter in an array. These are drawn based on the seir::parameters section of the config, passed in as p_config.
-        :param n_days: number of time interval
-        :param nsubpops: number of spatial nodes
-        :return:  array of shape (nparam, n_days, nsubpops) with all parameters for all nodes and all time (same value)
+        Format all parameters as a numpy array including sampling.
+        
+        The entries in the output array are filled based on the input given in the 
+        parameters section of a yaml config file. If the given parameter is pulled from
+        a distribution rather than fixed the values will be pulled from that 
+        distribution. If an appropriate value cannot be found for an entry then a 
+        `np.nan` is returned.
+        
+        Args:
+            n_days: The number of days to generate an array for.
+            nsubpops: The number of subpopulations to generate an array for.
+        
+        Returns:
+            A numpy array of size (`npar`, `n_days`, `nsubpops`) where `npar` 
+            corresponds to the `npar` attribute of this class. 
+            
+        Note:
+            If any of the parameters are 'timeseries' type parameters then `n_days` and
```

Comment:
Yes, you get an exception raised by numpy about shapes not being compatible on this line:
```python
param_arr[idx] = self.pdata[pn]["ts"].values
```
I suppose you could provide a single date time series and it would be fine since numpy could make those shapes compatible (although, the constructor doesn't allow missing dates for time series params). I decided to place this info under the `Notes` (I need to fix typo) section instead of the `Raises` since this method doesn't raise the exception itself.